File: lecture_002/factorial.java
// import java.util.Scanner;

// public class factorial{
//     public static Scanner scn = new Scanner(System.in);
//     public static void main(String[] args){
//         System.out.println("Enter number for factorial: ");
//         int n = scn.nextInt();
//         System.out.println("factorial: ");
//         System.out.println(facto(n));
//     }



// public static int facto(int n){
//     int ans =1;
//     for(int i=1;i<=n;i++){
//         ans*=i;
//     }
//     return ans;
// }

// }

//AT HOME

import java.util.Scanner;

public class factorial{
    public static Scanner scn = new Scanner(System.in);
    public static void main(String[] args){
        System.out.println("Enter number for factorial..");
        int n= scn.nextInt();
        facto(n);
    }

    public static int facto(int n) {
        int fact = 1;
        for(int i=1;i<=n;i++){
            fact*=i;
        }
        System.out.println("factorial is : " + fact);
        
        return 0 ;
    }
}

File: lecture_002/factorial.cpp
// #include<iostream>

// using namespace std;


// int facto(int n){
//     int ans =1;
//     for(int i=1;i<=n;i++){
//         ans*=i;
//     }
//     return ans;
// }

// int main(int args, char **argv){
//     int n;
//     cout<<"Enter number for factorial";
//     cin>>n;
//     cout<<"factorial :"<<facto(n);
//     return 0;
// }


//AT HOME
#include<iostream>

using namespace std;

int facto(int n){
    int fact = 1;
    for(int i =1;i<=n;i++){
        fact*=i;
    }
    return fact;
}
int nCr(int n, int r){
    int ncr = facto(n)/(facto(r)*facto(n-r));
    cout<<"The nCr is : "<<ncr;
}



int main(){
    int n,r;
    char ans;
    cout<<"Enter number: ";
    cin>>n;
    cout<<"The factorial is : "<<facto(n);
    cout<<"\nDo you wish to find nCr..? Say (y/n)";
    cin>>ans;
    if(ans=='y'){
        cout<<"Give the value of r: ";
        cin>>r;
        nCr(n,r);
    }
}